web/src/js/components/ProxyApp.tsx

```diff
@@ -77,13 +77,28 @@ class ProxyAppMain extends Component<ProxyAppMainProps, ProxyAppMainState> {
     };
 
     componentDidMount() {
-        window.addEventListener("keydown", this.props.onKeyDown);
+        //window.addEventListener("keydown", this.props.onKeyDown);
+        this.updateKeyDownListener(this.state.ActiveMenu);
+    }
+
+    componentDidUpdate(_, prevState: ProxyAppMainState) {
+        if (prevState.ActiveMenu !== this.state.ActiveMenu) {
+            this.updateKeyDownListener(this.state.ActiveMenu);
+        }
```

Comment:
I'm not sure if I understood it correctly. Do we have a keyboard duck? I only see a `Keyboard.tsx`. How can I access the current screen?
P.S. I added this because I couldn't type anything in the input field for the local applications. This was the only solution I found, even though the input will probably disappear in the future.

Reply:
It is a bit of a hack, but we have https://github.com/mitmproxy/mitmproxy/blob/main/web/src/js/ducks/ui/keyboard.tsx - I guess it's not a proper duck. 😛 If this is catching events, we can maybe call `e.stopPropagation` similar to how we do it [here](https://github.com/mitmproxy/mitmproxy/blob/95bf441bc3a1dfcac2cb313cac8e72d3716a4bed/web/src/js/components/editors/ValueEditor.tsx#L159). Does that make more sense?

Reply:
I see thank you, I'll figure that out :)